File: 프로그래머스/2/64065. 튜플/튜플.js
function solution(s) {
    let answer = [];
    
    let parts = s.slice(2, -2).split('},{');
    
    parts.sort((a, b) => a.length - b.length);
    
    let sets = parts.map(str => str.split(',').map(Number));
    
    for (let currentSet of sets) {
        for (let value of answer) {
            let index = currentSet.indexOf(value);
            if (index !== -1) {
                currentSet.splice(index, 1);
            }
        }
        answer.push(currentSet[0]);
    }
    
    return answer;
}

File: 프로그래머스/2/64065. 튜플/튜플.py
def solution(s):
    answer = []
    s = s[2:-2].split('},{')
    s = sorted(s, key=lambda x: len(x))
    
    s = [list(map(int, tup.split(','))) for tup in s]
    
    for tup in s:
        for i in answer:
            if i in tup:
                tup.remove(i)
        answer.append(tup[0])
        
    return answer